src/Foundation/Commerce/code/Infrastructure/Pipelines/Login/GetCommerceUserProcessor.cs

```diff
@@ -0,0 +1,73 @@
+﻿//    Copyright 2020 EPAM Systems, Inc.
+// 
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0
+// 
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+namespace Wooli.Foundation.Commerce.Infrastructure.Pipelines.Login
+{
+    using System.Web.Security;
+
+    using Account.Infrastructure.Pipelines.Login;
+
+    using Base.Infrastructure.Pipelines;
+
+    using ModelMappers.Users;
+
+    using Providers;
+
+    using Sitecore.Diagnostics;
+
+    public class GetCommerceUserProcessor : PipelineProcessor<LoginPipelineArgs>
+    {
+        private readonly ICustomerProvider customerProvider;
+
+        private readonly IUserMapper userMapper;
+
+        public GetCommerceUserProcessor(ICustomerProvider customerProvider, IUserMapper userMapper)
+        {
+            Assert.ArgumentNotNull(customerProvider, nameof(customerProvider));
+            Assert.ArgumentNotNull(userMapper, nameof(userMapper));
+
+            this.customerProvider = customerProvider;
+            this.userMapper = userMapper;
+        }
+
+        public override void Process(LoginPipelineArgs args)
```

Comment:
- All the processors should be covered by unit tests. 

The following comments can be implemented in scope of the next request:
- Hide the logic of getting username under `this.customerProvider.GetCommerceUser`. For that create 2 methods in the customerProvider interface: `GetUser(string email)` and `GetUser(Guid contactId)`. Make `GetCommerceUser` obsolete, or remove if no references
- Do not forget about unit tests